Lib/test/test_unpack.py

```diff
@@ -135,13 +144,57 @@
       ...
     TypeError: cannot unpack non-iterable int object
 
-Unpacking to an empty iterable should raise ValueError
+Unpacking to an empty iterable should raise UnpackError
 
     >>> () = [42]
     Traceback (most recent call last):
       ...
-    ValueError: too many values to unpack (expected 0)
+    UnpackError: too many values to unpack (expected 0, got 1)
+
+Unpacking to an empty iterable should raise UnpackError, but it won't consume the
+iterable if it doesn't have a pre-determined length
+
+    >>> it = iter(range(100))
+    >>> x, y, z = it
+    Traceback (most recent call last):
+      ...
+    UnpackError: too many values to unpack (expected 3)
+    >>> next(it)
+    4
+
+Unpacking unbalanced dict
+
+    >>> d = {4: 'four', 5: 'five', 6: 'six', 7: 'seven'}
+    >>> a, b, c = d
+    Traceback (most recent call last):
+      ...
+    UnpackError: too many values to unpack (expected 3, got 4)
 
+Ensure that `__len__()` is respected when showing the error message
```

Comment:
```suggestion
Ensure that __len__() is respected when showing the error message
```

We are in a non-rst/md context so no need for rendering.

Reply:
IMO, Markdown-style formatting is quite readable and works great in docstrings/doctests.

Reply:
Fine by me.